src/components/metadata-modal/metadata-modal.js

```diff
@@ -10,12 +11,16 @@ import getShortType from '../../utils/short-type';
 import { getClickedNodeMetaData } from '../../selectors/metadata';
 import './metadata-modal.scss';
 
-const MetadataModal = ({ metadata, onToggle, visible }) => {
+const MetadataModal = ({ metadata, onToggle, visible, theme }) => {
   const hasPlot = metadata?.previewType === 'PlotlyPreview';
   const hasImage = metadata?.previewType === 'ImagePreview';
   const hasTable = metadata?.previewType === 'TablePreview';
+  const hasJSON = metadata?.previewType === 'JSONPreview';
 
-  if (!visible.metadataModal || (!hasPlot && !hasImage && !hasTable)) {
+  if (
+    !visible.metadataModal ||
+    (!hasPlot && !hasImage && !hasTable && !hasJSON)
```

Comment:
[nit] may be for readability, we can combine `!hasPlot && !hasImage && !hasTable && !hasJSON` to something like `hasMetadataContent = hasPlot || hasImage || hasTable || hasJSON` and use it in the if condition